client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java

```diff
@@ -405,9 +405,27 @@ public final boolean isRegistrationAssertionRequired() {
         details.put(TracingHelper.TAG_QOS.getKey(), sender.getQoS().toString());
         currentSpan.log(details);
 
+        final Long timerId = config.getRequestTimeout() > 0
```

Comment:
I wonder if it wouldn't be better to start the timer only *after* the message has actually been sent (or at least been put into the outbound queue of the Proton-J stack). This would prevent a *false negative* (i.e. the timer task firing) in cases where the message ends up in the outbound queue and is sent with some delay only. WDYT?

Reply:
I think that wouldn't help since there is no way to start the timer exactly when the message is actually sent (ie. taken from the outbound queue).